armory/scenarios/image_classification.py

```diff
@@ -41,9 +49,9 @@ def _evaluate(self, config: dict) -> dict:
                 split_type="train",
                 preprocessing_fn=preprocessing_fn,
             )
-            if config["defense"] is not None:
-                logger.info("loading defense")
-                defense = load_defense(config["defense"], classifier)
+            if defense_type == "Trainer":
```

Comment:
So if you had a Transformer here, it would never get called by `load_defense_wrapper` and display the error. It would just run silently without the defense right?

Reply:
Correct.

I can have it try to load from the scenario and raise the NotImplementedError in the loading. That is more explicit. Thoughts?

Reply:
Yeah I would prefer that I think

Reply:
Updated. Let me know what you think.